app/views/layouts/application.html.erb

```diff
@@ -40,7 +40,9 @@
         <% end %>
         <% if user_signed_in? %>
     			<div class="buttons">
-  	  			<button class="button"><%= link_to "New Post", new_post_path %></button>
+            <%#- if policy_scope(@user.posts) %>
```

Comment:
Can you explain why you use a `policy_scope` here?

Reply:
Well I wasn't sure if I necessarily needed to use a scope but I was trying a few different approaches. My goal is to only render this button for users that are admins. Because the PostsController already restricts the create/edit/update/publish (all) actions to admin users. The problem seems to be that the New Post button is on my layout so depending on what partial form is rendered at the same time as the layout, the activeRecord object/collection is different. Sometimes there are multiple posts and sometimes it's singular. A simple solution would be to remove the button from my layout, but I'm going to try to find another solution. What do you think.

Reply:
I could just check if current_user.role == 1 which is admin for my application. what do you think?